EmailAddressMangler.ps1

```diff
@@ -414,293 +145,52 @@
-                        $FullUserList += "$lname.$fname"
-                    }
-                    elseif($AddressConvention -eq "ln-fn")
-                    {
-                        $fname = $fname.ToLower()
-                        $lname = $lname.ToLower()
-                        $FullUserList += "$lname-$fname"
-                    }
-                    elseif($AddressConvention -eq "lifn")
-                    {
-                        $fname = $fname.ToLower()
-                        $lname = $lname.ToLower()
-                        $finit = $fname.Substring(0,1)
-                        $FullUserList += "$linit$fname"
-                    }
-                    elseif($AddressConvention -eq "li-fn")
-                    {
-                        $fname = $fname.ToLower()
-                        $lname = $lname.ToLower()
-                        $linit = $lname.Substring(0,1)
-                        $FullUserList += "$linit-$fname"
-                    }
-                    elseif($AddressConvention -eq "li.fn")
-                    {
-                        $fname = $fname.ToLower()
-                        $lname = $lname.ToLower()
-                        $linit = $lname.Substring(0,1)
-                        $FullUserList += "$linit.$fname"
-                    }
-                    elseif($AddressConvention -eq "lnfi")
-                    {
-                        $fname = $fname.ToLower()
-                        $lname = $lname.ToLower()
-                        $finit = $fname.Substring(0,1)
-                        $FullUserList += "$lname$finit"
-                    }
-                    elseif($AddressConvention -eq "ln-fi")
-                    {
-                        $fname = $fname.ToLower()
-                        $lname = $lname.ToLower()
-                        $finit = $fname.Substring(0,1)
-                        $FullUserList += "$lname-$finit"
-                    }
-                    elseif($AddressConvention -eq "ln.fi")
-                    {
-                        $fname = $fname.ToLower()
-                        $lname = $lname.ToLower()
-                        $finit = $fname.Substring(0,1)
-                        $FullUserList += "$lname.$finit"
-                    }
-                    else
-                    {
-                    Write-Host -ForegroundColor "red" "[*] Please enter an email address naming convention with the -AddressConvention flag."
-                    Write-Host -ForegroundColor "red" "[*] Accepted formats include fnln, filn, fn.ln, fn etc. where fn = first name, ln = last name, fi = first initial, and li = last initial."
-                    return
-                    }
+                    $FullUserList += Format -Convention $AddressConvention -FirstName $fname -LastName $lname
                 }
             }
         }
     }
     $FullUserList = $FullUserList | sort | Get-Unique
     Write-Output $FullUserList
+}
+
+$Conventions = @{

```

Comment:
Highlighting the core change, the remove changes above were consolidated down to this hashtable.